### PR TITLE
iOS example with swift ui

### PR DESCRIPTION
**Pull request by bachittle**
Similar to what is found in the whisper.cpp repository, this is a working iOS example of running llama.cpp with a minimalist swift ui wrapper. It also uses ggml-metal so inference is very fast. It works with gguf v2 models. It is able to run starcoder-1b very well, but mistral-7b needs to be heavily quantized, and even then it should be running on an iPhone with 6-8gb ram, or else you will run into memory issues. 

https://github.com/bachittle/llama.cpp/assets/39804642/e290827a-4edb-4093-9642-2a5e399ec545
 
*gg: I added the video to the post because I think it is cool* 👍

### CODE REVIEW

File: examples/llama.swiftui/llama.cpp.swift/LibLlama.swift [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+    func completion_loop() -> String {
+        var new_token_id: llama_token = 0
+        
+        let n_vocab = llama_n_vocab(model)
+        let logits = llama_get_logits_ith(context, batch.n_tokens - 1)
+                
+        var candidates = Array<llama_token_data>()
+        candidates.reserveCapacity(Int(n_vocab))
+        
+        for token_id in 0...n_vocab {
```
Comment (aehlke) [inline examples/llama.swiftui/llama.cpp.swift/LibLlama.swift:107]: not `..<`? other C++ examples in the repo use `<`

Comment (aehlke) [reply examples/llama.swiftui/llama.cpp.swift/LibLlama.swift:107]: the rest looks good to me, thanks for this work

Comment (bachittle) [reply examples/llama.swiftui/llama.cpp.swift/LibLlama.swift]: fixed

**Comment by ggerganov** [comment]
Thanks! Could you add some basic CI workflow that builds the example?

**Comment by ggerganov** [comment]
Also, I think it's time to make an advanced version of the [teleprompter application](https://twitter.com/ggerganov/status/1605322535930941441) from last year 😄 

Anyone want to give it a try? 🤣

**Comment by aehlke** [comment]
I'm using this work in a fork, and it works well for me, thanks

**Comment by bachittle** [comment]
Made a minimal workflow. You can check this local PR to see if all checks passed before approving here: https://github.com/bachittle/llama.cpp/pull/5

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -497,6 +497,16 @@ jobs:
         with:
           path: |
             cudart-llama-bin-win-cu${{ matrix.cuda }}-x64.zip
+  ios-xcode-build:
```
Comment (ggerganov) [inline .github/workflows/build.yml:500]:
```suggestion

  ios-xcode-build:
```

**Review by ggerganov — APPROVED** [review]

**Pull request merged**